performance_manager/lib/postgres_utils.py

```diff
@@ -16,6 +16,69 @@
 HERE = os.path.dirname(os.path.abspath(__file__))
 
 
+def postgres_create_modified_trigger(
+    table: sa.schema.Table,
+    connection: sa.engine.Connection,
+    **_: Any,
+) -> None:
+    """
+    sqlalchemy will listen for new table create events and when detected,
+    run this function to suplement table creation by creating update_on
+    triggers for tables containing an 'update_on' field
+
+    this function will only be used for postgres db
+    """
+    update_column_name = "updated_on"
+    trigger_function_statement = sa.DDL(
+        f"CREATE OR REPLACE FUNCTION update_modified_columns() "
+        f"RETURNS TRIGGER AS $$ "
+        f"BEGIN "
+        f"NEW.{update_column_name} = CURRENT_TIMESTAMP; "
+        f"RETURN NEW; "
+        f"END; $$ LANGUAGE plpgsql"
+    )
+    connection.execute(trigger_function_statement)
```

Comment:
postgres requires a trigger function to perform a trigger action, this trigger function will update the `updated_on` column with CURRENT_TIMESTAMP for all records provided